dataservice/api/family_relationship/models.py

```diff
@@ -35,62 +35,101 @@ class FamilyRelationship(db.Model, Base):
         backref=db.backref('incoming_family_relationships',
                            cascade='all, delete-orphan'))
 
     @classmethod
-    def query_all_relationships(cls, participant_kf_id):
-        """
-        Find all relationship records given a the kf_id of a particpant
+    def query_all_relationships(cls, participant_kf_id=None,
+                                model_filter_params=None):
         """
+        Find all family relationships for a participant
+
+        :param participant_kf_id: Kids First ID of the participant
+        :param model_filter_params: Filter parameters to the query
 
-        q = cls.query.filter(or_(
-            FamilyRelationship.participant_id == participant_kf_id,
-            FamilyRelationship.relative_id == participant_kf_id))
+        Given a participant's kf_id, return all of the biological
+        family relationships of the participant and the relationships
+        of the participant's family members.
+
+        If the participant does not have a family defined, then return
+        all of the immediate/direct family relationships of the participant.
+        """
+        # Apply model property filter params
+        if model_filter_params is None:
+            model_filter_params = {}
+        q = FamilyRelationship.query.filter_by(**model_filter_params)
+
+        # Get family relationships and join with participants
+        q = q.join(Participant, or_(FamilyRelationship.participant1,
+                                    FamilyRelationship.participant2))
+
+        # Do this bc query.get() errors out if passed None
+        if participant_kf_id:
+            pt = Participant.query.get(participant_kf_id)
+            family_id = pt.family_id if pt else None
+
+            # Use family to get all family relationships in participants family
+            if family_id:
+                q = q.filter(Participant.family_id == family_id)
+
+            # No family provided, use just family relationships
+            # to get only immediate family relationships for participant
+            else:
+                q = q.filter(or_(
+                    FamilyRelationship.participant1_id == participant_kf_id,
+                    FamilyRelationship.participant2_id == participant_kf_id))
+
+        # Don't want duplicates - return unique family relationships
+        q = q.group_by(FamilyRelationship.kf_id)
 
         return q
 
     def __repr__(self):
-        return '<{} is {} of {}>'.format(self.participant,
-                                         self.participant_to_relative_relation,
-                                         self.relative)
+        return '<{} is {} of {}>'.format(
+            self.participant1.kf_id,
+            self.participant1_to_participant2_relation,
+            self.participant2.kf_id)
 
 
-@event.listens_for(FamilyRelationship.participant_to_relative_relation, 'set')
+@event.listens_for(FamilyRelationship.participant1_to_participant2_relation,
```

Comment:
Perhaps generalize this for both `participant1_to_participant2_relation` and `participant2_to_participant1_relation`?